src/main/java/org/folio/linked/data/configuration/kafka/KafkaProducerConfiguration.java

```diff
@@ -34,7 +36,7 @@ public FolioMessageProducer<ResourceIndexEvent> bibliographicMessageProducer(
   ) {
     var producer = new FolioMessageProducer<>(resourceIndexEventMessageTemplate,
       linkedDataTopicProperties::getWorkSearchIndex);
-    producer.setKeyMapper(ResourceIndexEvent::getId);
+    producer.setKeyMapper(rie -> ((LinkedDataWork) rie.getNew()).getId());
```

Comment:
Changes are for alignment with common folio approach: eventId is random UUID;
And here we take resource id instead of it, to keep our approach: message key is resource id